src/libraries/System.Threading.Tasks.Dataflow/src/Blocks/TransformManyBlock.IAsyncEnumerable.cs

```diff
@@ -0,0 +1,247 @@
+            TargetCore<TInput> target = _target;
+            bool isBounded = target.IsBounded;
+
+            // Handle invalid items (null enumerables) by delegating to the base
+            if (item == null)
+            {
+                _reorderingBuffer.AddItem(id, null, false);
+                if (isBounded) target.ChangeBoundingCount(count: -1);
+                return;
+            }
+
+            // Determine whether this id is the next item, and if it is and if we have a trusted list,
+            // try to output it immediately on the fast path.  If it can be output, we're done.
+            // Otherwise, make forward progress based on whether we're next in line.
+            bool? isNextNullable = _reorderingBuffer.AddItemIfNextAndTrusted(id, null, false);
+            if (!isNextNullable.HasValue) return; // data was successfully output
+            bool isNextItem = isNextNullable.Value;
+
+            // By this point, either we're not the next item, in which case we need to make a copy of the
+            // data and store it, or we are the next item and can store it immediately but we need to enumerate
+            // the items and store them individually because we don't want to enumerate while holding a lock.
+            List<TOutput>? itemCopy = null;
+            try
+            {
+                // If this is the next item, we can output it now.
+                if (isNextItem)
+                {
+                    await StoreOutputItemsNonReorderedWithIterationAsync(item).ConfigureAwait(false);
+                    // here itemCopy remains null, so that base.AddItem will finish our interactions with the reordering buffer
+                }
+                else
+                {
+                    // We're not the next item, and we're not trusted, so copy the data into a list.
+                    // We need to enumerate outside of the lock in the base class.
+                    int itemCount = 0;
+                    try
+                    {
+                        itemCopy = new List<TOutput>();
+                        await foreach (TOutput element in item.ConfigureAwait(false))
+                        {
+                            itemCopy.Add(element);
+                        }
+                        itemCount = itemCopy.Count;
+                    }
+                    finally
+                    {
+                        // If we're here successfully, then itemCount is the number of output items
+                        // we actually received, and we should update the bounding count with it.
+                        // If we're here because ToList threw an exception, then itemCount will be 0,
+                        // and we still need to update the bounding count with this in order to counteract
+                        // the increased bounding count for the corresponding input.
+                        if (isBounded) UpdateBoundingCountWithOutputCount(count: itemCount);
```

Comment:
Nit: Add Braces

*Personally, I also prefer the style in such cases. But the runtime coding style specifies that you should use braces.*
https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/coding-style.md (18.)